designs/SearchCommandSpec.md

```diff
@@ -0,0 +1,98 @@
+
+## Solution
+
+Th search feature leverages the search service provided by the NuGet API. Search targets multiple sources and displays all of their results one after the other. The command line interface is simple: __./NuGet.exe search \<query\>__.
+```
+PS C:\> ./NuGet.exe search logging
+```
+
+Each search result displays the name, version, number of downloads, and description of the package. Additionally, Search uses pagination to provide a clean viewing experience for users as they look at the results of their search. This ensures that users have access to the search results in order of decreasing relevance. A single page of results is shown at a time, and the user can look at the next page by pressing the __space__ key.
+
+```
+====================
+Source: NuGet.org
+--------------------
+> Microsoft.Extensions.Logging.Abstractions | v5.0.0-preview.6.20305.6 | DLs: 345145935
+Logging abstractions for Microsoft.Extensions.Logging.
+
+Commonly Used Types:
+Microsoft.Extensions.Logging.ILogger
+Microsoft.Extensions.Logging.ILoggerFactory
+Microsoft.Extensions.Logging.ILogger&lt;TCategoryName&gt;
+Microsoft.Extensions.Logging.LogLevel
+Microsoft.Extensions.Logging.Logger&lt;T&gt;
+Microsoft.Extensions.Logging.LoggerMessage
+Microsoft.Extensions.Logging.Abstractions.NullLogger
+
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging | v5.0.0-preview.6.20305.6 | DLs: 243186566
+Logging infrastructure default implementation for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.IdentityModel.Logging | v6.6.0 | DLs: 120177247
+Includes Event Source based logging support.
+
+--------------------
+> Microsoft.Extensions.Logging.Configuration | v5.0.0-preview.6.20305.6 | DLs: 63205816
+Configuration support for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging.Console | v5.0.0-preview.6.20305.6 | DLs: 61796431
+Console logger provider implementation for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging.Debug | v5.0.0-preview.6.20305.6 | DLs: 58643995
+Debug output logger provider implementation for Microsoft.Extensions.Logging. This logger logs messages to a debugger monitor by writing messages with System.Diagnostics.Debug.WriteLine().
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging.EventSource | v5.0.0-preview.6.20305.6 | DLs: 32121683
+EventSource/EventListener logger provider implementation for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Serilog.Extensions.Logging | v3.0.2-dev-10280 | DLs: 43115462
+-- More  --
```

Comment:
You can probably cut this example down to 2-3 entries instead of the full actual results. This takes up a lot of space!

Reply:
I've made this change in my latest commit.